x-pack/plugins/osquery/server/lib/telemetry/helpers.ts

```diff
@@ -0,0 +1,156 @@
+
+  if (moment(executeTo).diff(lastExecutionTimestamp, 'hours') >= 24) {
+    return moment(executeTo).subtract(24, 'hours').toISOString();
+  }
+
+  return lastExecutionTimestamp;
+};
+
+/**
+ * Chunks an Array<T> into an Array<Array<T>>
+ * This is to prevent overloading the telemetry channel + user resources
+ *
+ * @param telemetryRecords
+ * @param batchSize
+ * @returns the batch of records
+ */
+export const batchTelemetryRecords = (
+  telemetryRecords: unknown[],
+  batchSize: number
+): unknown[][] =>
+  [...Array(Math.ceil(telemetryRecords.length / batchSize))].map(() =>
+    telemetryRecords.splice(0, batchSize)
+  );
+
+/**
+ * User defined type guard for PackagePolicy
+ *
+ * @param data the union type of package policies
+ * @returns type confirmation
+ */
+export function isPackagePolicyList(
+  data: string[] | PackagePolicy[] | undefined
+): data is PackagePolicy[] {
+  if (data === undefined || data.length < 1) {
+    return false;
+  }
+
+  return (data as PackagePolicy[])[0].inputs !== undefined;
+}
+
+/**
+ * Constructs the packs telemetry schema from a collection of packs saved objects
+ */
+export const templatePacks = (
+  packsData: SavedObjectsFindResponse['saved_objects'],
+  clusterInfo: ESClusterInfo,
+  licenseInfo: ESLicense | undefined
+) =>
+  packsData.map((item) => {
+    const template: ListTemplate = {
+      '@timestamp': moment().toISOString(),
+      cluster_uuid: clusterInfo.cluster_uuid,
+      cluster_name: clusterInfo.cluster_name,
+      license_id: licenseInfo?.uid,
+    };
+
+    // cast exception list type to a TelemetryEvent for allowlist filtering
+    const filteredPackItem = copyAllowlistedFields(
+      packEventFields,
+      item.attributes as unknown as TelemetryEvent
+    );
+
+    return {
+      ...template,
+      id: item.id,
+      ...filteredPackItem,
+    };
+  });
+
+/**
+ * Constructs the packs telemetry schema from a collection of packs saved objects
+ */
+export const templateSavedQueries = (
+  savedQueriesData: SavedObjectsFindResponse['saved_objects'],
+  clusterInfo: ESClusterInfo,
+  licenseInfo: ESLicense | undefined
+) =>
+  savedQueriesData.map((item) => {
+    const template: ListTemplate = {
+      '@timestamp': moment().toISOString(),
+      cluster_uuid: clusterInfo.cluster_uuid,
+      cluster_name: clusterInfo.cluster_name,
+      license_id: licenseInfo?.uid,
+    };
+
+    // cast exception list type to a TelemetryEvent for allowlist filtering
+    const filteredSavedQueryItem = copyAllowlistedFields(
+      savedQueryEventFields,
+      item.attributes as unknown as TelemetryEvent
+    );
+
+    return {
+      ...template,
+      id: item.id,
+      ...filteredSavedQueryItem,
+    };
+  });
+
+/**
+ * Convert counter label list to kebab case
+ *
+ * @param label_list the list of labels to create standardized UsageCounter from
+ * @returns a string label for usage in the UsageCounter
+ */
+export function createUsageCounterLabel(labelList: string[]): string {
```

Comment:
You don't need this func unless you are going to count events like we do (query documents dropped etc)